cognite/client/data_classes/relationships.py

```diff
@@ -1,139 +1,129 @@
         cognite_client=None,
     ):
-        self.source = source
-        self.target = target
+        self.external_id = external_id
+        self.source_external_id = source_external_id
+        self.source_type = source_type
+        self.target_external_id = target_external_id
+        self.target_type = target_type
         self.start_time = start_time
         self.end_time = end_time
         self.confidence = confidence
-        self.data_set = data_set
-        self.external_id = external_id
-        self.relationship_type = relationship_type
+        self.data_set_id = data_set_id
         self.created_time = created_time
         self.last_updated_time = last_updated_time
+        self.labels = labels
         self._cognite_client = cognite_client
 
-    def _copy_resolve_targets(self):
+    def _validate_resource_types(self):
         rel = copy.copy(self)
-        rel.source = self._resolve_target(rel.source)
-        rel.target = self._resolve_target(rel.target)
+        self._validate_target(rel.source_type)
+        self._validate_target(rel.target_type)
         return rel
 
     @staticmethod
-    def _resolve_target(target):
+    def _validate_target(target):
         if isinstance(target, dict) or target is None:
             return target
 
-        from cognite.client.data_classes import Asset, Event, FileMetadata, TimeSeries, Sequence
-
-        _TARGET_TYPES = {
-            Asset: "Asset",
-            TimeSeries: "TimeSeries",
-            FileMetadata: "File",
-            Event: "Event",
-            Sequence: "Sequence",
-        }
-        typestr = _TARGET_TYPES.get(target.__class__)
-        if typestr:
-            return {"resource": typestr, "resourceId": target.external_id}
-        raise ValueError("Invalid source or target '{}' of type {} in relationship".format(target, target.__class__))
+        _TARGET_TYPES = {"asset", "timeseries", "file", "event", "sequence"}
+        if target.lower() not in _TARGET_TYPES:
+            raise ValueError("Invalid source or target '{}' in relationship".format(target))
 
 
 class RelationshipFilter(CogniteFilter):
-    """Filter on relationships with exact match. Multiple filter elments in one property, e.g. `dataSets: [ "a", "b" ]`, will return all relationships where the dataSet field is either `a` or `b`. Filters in multiple properties will return the relationships that match all criteria. Filters on a `resourceId` without a `resource` (type) in sources and targets will return relationships that match the resourceId and match any resource type. If the filter is not specified it default to an empty filter.
+    """Filter on relationships with exact match. Multiple filter elments in one property, e.g. sourceExternalIds: [ "a", "b" ], will return all relationships where the sourceExternalId field is either a or b. Filters in multiple properties will return the relationships that match all criteria. If the filter is not specified it default to an empty filter.
```

Comment:
weird documentation change - the previous one was autogenerated so changing it to a different non-sdk related one doesn't make sense.

Reply:
I just copied the service contract here. Should this be reverted?

Reply:
I would make the docstring relevant python code `e.g. RelationshipFilter( .... )`